File: vantage6-node/joey/server/models/organization.py
from sqlalchemy import Column, String
from sqlalchemy.orm import Session, relationship
from sqlalchemy.orm.exc import NoResultFound

from .base import Base, Database
from .member import Member
from .collaboration import Collaboration
from .user import User


class Organization(Base):
    """A legal entity.
    
    An organization plays a central role in managing distributed tasks. Each
    Organization contains a public key which other organizations can use to 
    send encrypted messages that only this organization can read.
    """

    # fields
    name = Column(String)
    domain = Column(String)
    address1 = Column(String)
    address2 = Column(String)
    zipcode = Column(String)
    country = Column(String)
    public_key = Column(String)

    # relations
    collaborations = relationship("Collaboration", secondary="Member",
        back_populates="organizations")
    task_assignments = relationship("TaskAssignment", 
        back_populates="organization")
    nodes = relationship("Node", back_populates="organization")
    users = relationship("User", back_populates="organization")

    @classmethod
    def get_by_name(cls, name):
        session = Database().Session
        try:
            return session.query(cls).filter_by(name=name).first()
        except NoResultFound:
            return None

    def __repr__(self):
        number_of_users = len(self.users)
        return ("<Organization "
            f"name:{self.name}, "
            f"domain:{self.domain}, "
            f"users:{number_of_users}"
        ">")

File: vantage6-ui/src/app/organization/interfaces/organization.ts
import { Node } from 'src/app/node/interfaces/node';
import { Resource } from 'src/app/shared/enum';

export interface Organization {
  id: number;
  type: string;
  name: string;
  address1: string;
  address2: string;
  zipcode: string;
  country: string;
  domain: string;
  public_key: string;
  is_being_created?: boolean;
}

export interface OrganizationInCollaboration extends Organization {
  node?: Node;
}

export const EMPTY_ORGANIZATION = {
  id: -1,
  type: Resource.ORGANIZATION,
  name: '',
  address1: '',
  address2: '',
  zipcode: '',
  country: '',
  domain: '',
  public_key: '',
};
